src/audio/host-zephyr.c

```diff
@@ -454,9 +455,20 @@ static int host_copy_normal(struct host_data *hd, struct comp_dev *dev)
 	};
 	notifier_event(hd->chan, NOTIFIER_ID_DMA_COPY,
 		       NOTIFIER_TARGET_CORE_LOCAL, &next, sizeof(next));
-	ret = dma_reload(hd->chan->dma->z_dev, hd->chan->index, 0, 0, copy_bytes);
-	if (ret < 0)
-		comp_err(dev, "host_copy_normal(): dma_copy() failed, ret = %u", ret);
+
+	hd->partial_size += copy_bytes;
+	buffer_c = buffer_acquire(hd->dma_buffer);
+
+	/* Reload DMA when half of the buffer has been used */
+	if (hd->partial_size >= buffer_c->stream.size >> 1) {
```

Comment:
@lyakh seems like a good idea to reduce DMA traffic but do you think it makes sense to limit this to playback only?

Reply:
@ranj063 what effect would this have for capture? If the host buffer is large, it would delay triggering the host until more data is available, right? So, that means some additional latency. Do we use such larger buffers for capture? If we don't - that isn't a problem, if we do - isn't that the reason why we use them - to buffer more data in them before waking up the consumer?

Reply:
I don't think we use large buffers for capture yet but I think there may be a plan to use 10ms buffers if not larger. But yes in that case, maybe it makes sense to do the same thing there as well.

Reply:
BTW @lyakh do we know that the DMA transfer block size is equal to the size of the DMA buffer size in the case of deep buffer? It should be but just wondering if it really is or not.

Reply:
I think in my tests I saw the deep buffer size equal to 38400 bytes, i.e. 100 * 384, and 384 which is 32 bits * 2 channels * 48 and corresponds to 1ms at 48000 samples per second, i.e. it's a 100ms buffer, which seems correct. And I hope that isn't the size that the DMA transfers in one go? Or have I misunderstood?